docs/explanation/core/proposer-based-timestamps.md

```diff
@@ -55,44 +33,182 @@ The following protocols and application features require a reliable source of ti
+when instantiating a new network or when upgrading an existing a network that
+uses BFT Time.
+
+Consensus parameters are configured through the genesis file, for new chains, or by the ABCI application, for new and existing chains, and are the same
+across all nodes in the network.
+
+### `SynchronyParams.Precision`
+
+The `Precision` parameter configures the acceptable upper-bound of clock drift
+among all of the validators in the network.
+Any two validators are expected to have clocks that differ by at most
+`Precision` at any given instant.
+
+The `Precision` parameter is of [`time.Duration`](https://pkg.go.dev/time#Duration) type.
+
+Networks should choose a `Precision` that is large enough to represent the
+worst-case for the clock drift among all participants.
+Due to the [leap second events](https://github.com/tendermint/tendermint/issues/7724),
+it is recommended to use set `Precision` to at least `500ms`.
+
+### `SynchronyParams.MessageDelay`
+
+The `MessageDelay` parameter configures the acceptable upper-bound for the
+end-to-end delay for transmitting a `Proposal` message from the proposer to
+_all_ validators in the network.
+
+The `MessageDelay` parameter is of [`time.Duration`](https://pkg.go.dev/time#Duration) type.
+
+Networks should choose a `MessageDelay` that is large enough to represent the
+delay for a `Proposal` message to reach all participants.
+As `Proposal` messages are fixed-size, this delay should not depend, a priori,
+on the size of proposed blocks.
+But it does depend on the number of nodes in the network and latency of their
+connections.
+
+### `FeatureParams.PbtsEnableHeight`
+
+The `PbtsEnableHeight` parameter configures the first height at which the PBTS
+algorithm should be adopted for generating and validating block timestamps in a network.
+
+The `PbtsEnableHeight` parameter is an integer.
+
+While `PbtsEnableHeight` is set to `0`, the network will adopt the legacy BFT
+Time algorithm.
+
+When `PbtsEnableHeight` is set to a height `H > 0`, the network will switch to
+the PBTS algorithm from height `H`.
+The network will still adopt the legacy BFT Time algorithm to produce and
+validate block timestamps until height `H - 1`.
+The enable height `H` must be a future height, i.e., larger than the current
+blockchain height.
+
+Once `PbtsEnableHeight` is set and the PBTS algorithm is enabled (i.e., from height
+`PbtsEnableHeight`), it is not possible to return to the legacy BFT Time algorithm.
+The switch to PBTS is therefore irreversible.
+
+Finally, if `PbtsEnableHeight` is set `1` in the genesis file or by the application
+upon the ABCI `InitChain` method, the network will adopt PBTS from the first
+height. This is the recommended setup for new chains.
+
+
+## Important Notes
+
+When configuring a network to adopt the PBTS algorithm, the follow steps must be considered:
+
+1. Make sure that the clocks of validators are [synchronized](#clock-synchronization) **before** enabling PBTS.
```

Comment:
Numbering here is messed up

Reply:
Why? This helps referring to a specific observation, as done in following paragraph.